src/field_theory/minimal_polynomial.lean

```diff
@@ -0,0 +1,193 @@
+/-
+Copyright (c) 2019 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes, Johan Commelin
+-/
+
+import ring_theory.integral_closure
+
+universes u v w
+open polynomial set function
+
+noncomputable theory
+
+/- Turn down the instance priority for subtype.decidable_eq and use classical.dec_eq everywhere,
+  to avoid diamonds -/
+local attribute [instance, priority 0] subtype.decidable_eq
+
+variables {α : Type u} {β : Type v}
+
+section min_poly_def
+variables [decidable_eq α] [decidable_eq β] [comm_ring α] [comm_ring β] [algebra α β]
```

Comment:
Why have `decidable_eq` arguments if this is all classical anyway?

Reply:
Fixed